_overviews/scala3-book/taste-indentation-significant-syntax.md

```diff
@@ -0,0 +1,189 @@
+tokens at certain line breaks. Grammatically, pairs of `<indent>` and `<outdent>` tokens have the same effect as pairs of braces `{` and `}`.
+
+The algorithm makes use of a stack `IW` of previously encountered indentation widths. The stack initially holds a single element with a zero indentation width. The _current indentation width_ is the indentation width of the top of the stack.
+
+There are two rules:
+
+1. An `<indent>` is inserted at a line break, if
+
+    - An indentation region can start at the current position in the source, and
+    - the first token on the next line has an indentation width strictly greater
+      than the current indentation width
+
+   An indentation region can start
+
+    - after the leading parameters of an `extension`, or
+    - after a `with` in a given instance, or
+    - after a `:` at the start of a template body (see discussion of `<colon>` below), or
+    - after one of the following tokens:
+
+      ```
+      =  =>  ?=>  <-  catch  do  else  finally  for
+      if  match  return  then  throw  try  while  yield
+      ```
+
+    - after the closing `)` of a condition in an old-style `if` or `while`.
+    - after the closing `)` or `}` of the enumerations of an old-style `for` loop without a `do`.
+
+   If an `<indent>` is inserted, the indentation width of the token on the next line
+   is pushed onto `IW`, which makes it the new current indentation width.
+
+2. An `<outdent>` is inserted at a line break, if
+
+    - the first token on the next line has an indentation width strictly less
+      than the current indentation width, and
+    - the last token on the previous line is not one of the following tokens
+      which indicate that the previous statement continues:
+      ```
+      then  else  do  catch  finally  yield  match
+      ```
+    - if the first token on the next line is a
+      [leading infix operator](../changed-features/operators.md).
+      then its indentation width is less then the current indentation width,
+      and it either matches a previous indentation width or is also less
+      than the enclosing indentation width.
+
+   If an `<outdent>` is inserted, the top element is popped from `IW`.
+   If the indentation width of the token on the next line is still less than the new current indentation width, step (2) repeats. Therefore, several `<outdent>` tokens
+   may be inserted in a row.
+
+   The following two additional rules support parsing of legacy code with ad-hoc layout. They might be withdrawn in future language versions:
+
+    - An `<outdent>` is also inserted if the next token following a statement sequence starting with an `<indent>` closes an indentation region, i.e. is one of `then`, `else`, `do`, `catch`, `finally`, `yield`, `}`, `)`, `]` or `case`.
+
+    - An `<outdent>` is finally inserted in front of a comma that follows a statement sequence starting with an `<indent>` if the indented region is itself enclosed in parentheses.
+
```

Comment:
The preceding is more like spec. I can't review that ATM, but I think some guidance about "when do I need to think about indentation" would be worthwhile. That is, normally, it should just work. When does it not just work?